lib/util/cache/package/decorator.ts

```diff
@@ -0,0 +1,126 @@
+import is from '@sindresorhus/is';
+import { logger } from '../../../logger';
+import * as packageCache from '.';
+
+type Handler<T> = (parameters: DecoratorParameters<T>) => Promise<unknown>;
+type Method<T> = (this: T, ...args: any[]) => any;
+type Decorator<T> = <U extends T>(
+  target: U,
+  key: keyof any,
```

Comment:
```suggestion
  key: keyof U,
```
shouldn't this a member of target type?

Reply:
Yes, I think this was leftover from when I was still hacking it all together!